esp32s3-hal/ld/memory.x

```diff
@@ -11,14 +11,27 @@ RESERVE_RTC_SLOW = 0;
 /* define stack size for both cores */
 STACK_SIZE = 8k;
 
-/* Specify main memory areas */
+VECTORS_SIZE = 0x400;
+
+/* Specify main memory areas
+
+ 40370000 <- IRAM/Icache -> 40378000 <- D/IRAM (I) -> 403E0000
+                            3FC88000 <- D/IRAM (D) -> 3FCF0000 <- DRAM/DCache -> 3FD00000
+
+ Startup code uses the IRAM from 0x403B9000 to 0x403E0000, which is not available for static
+ memory, but can only be used after app starts.
+
+ D cache use the memory from high address, so when it's configured to 16K/32K, the region
+ 0x3FCF000 ~ (3FD00000 - DATA_CACHE_SIZE) should be available. This region is not used as
+ static memory, leaving to the heap.
+*/
 MEMORY
 {
-  vectors_seg ( RX )     : ORIGIN = 0x40370000 + RESERVE_DRAM, len =  1k
-  iram_seg ( RX )        : ORIGIN = 0x40370400 + RESERVE_DRAM, len = 328k - 0x400
+  vectors_seg ( RX )     : ORIGIN = 0x40370000 + RESERVE_ICACHES, len =  1k
+  iram_seg ( RX )        : ORIGIN = 0x40370000 + RESERVE_ICACHES + VECTORS_SIZE, len = 328k - VECTORS_SIZE - RESERVE_ICACHES
+  dram_seg ( RW )        : ORIGIN = 0x3FC80000 + RESERVE_ICACHES + VECTORS_SIZE, len = 328k - VECTORS_SIZE - RESERVE_ICACHES
 
-  dram_seg ( RW )        : ORIGIN = 0x3FC80000 + RESERVE_DRAM, len = 328k - RESERVE_DRAM
-  reserved_for_boot_seg  : ORIGIN = 0x3FFDC200, len = 144k /* ???? SRAM1; reserved for static ROM usage; can be used for heap */
+  reserved_for_boot_seg  : ORIGIN = 0x3FFDC200, len = 144k /* reserved for static ROM usage; can be used for heap */
```

Comment:
```suggestion
  reserved_for_boot_seg  : ORIGIN = 0x3FCDC700, len = 0xB000 /* reserved for static ROM usage; can be used for heap */
```
**0xB000** is the sum of the IRAM region used for the Bootloader loader code and the DRAM region, which can be reclaimed after the Bootloader finishes loading the Application Image.